src/screens/AddNewItemScreen/index.tsx

```diff
@@ -26,6 +28,32 @@ export const AddNewItemScreen: FC = () => {
   const currentDate = `${date.getDate()}.${date.getMonth()}.${date.getFullYear()} at ${date.getHours()}:${date.getMinutes()}`;
   const buttonStyle = text ? style.button : style.buttonDis;
 
+  const goBack = () => {
+    navigation.goBack();
+  };
+
+  const onPressBack = () => {
+    if (text) {
+      return createAlertMessageHelper({
+        onPress: goBack,
+        message: `You have unsaved changes. Are you sure to discard them and leave the screen?`,
+        title: 'Discard changes',
+        cancelButtonTitle: "Don't leave",
+        confirmButtonTitle: 'Leave',
+      });
+    }
+
+    return goBack();
+  };
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerLeft: () => {
+        return <AddNewItemScreenBackButton onPress={onPressBack} />;
+      },
+    });
+  }, [onPressBack]);
```

Comment:
your onPressBack is created on every render 
use useCallback (check how times useEffect calls using console.log)